pkg/commands/git_commands/diff.go

```diff
@@ -40,3 +40,41 @@ func (self *DiffCommands) GetAllDiff(staged bool) (string, error) {
 			ToArgv(),
 	).RunWithOutput()
 }
+
+type DiffToolCmdOptions struct {
+	// The path to show a diff for. Pass "." for the entire repo.
+	Filepath string
+
+	// The commit against which to show the diff. Leave empty to show a diff of
+	// the working copy.
+	FromCommit string
+
+	// The commit to diff against FromCommit. Leave empty to diff the working
+	// copy against FromCommit. Leave both FromCommit and ToCommit empty to show
+	// the diff of the unstaged working copy changes against the index if Staged
+	// is false, or the staged changes against HEAD if Staged is true.
+	ToCommit string
+
+	// Whether to reverse the left and right sides of the diff.
+	Reverse bool
+
+	// Whether the given Filepath is a directory. We'll pass --dir-diff to
+	// git-difftool in that case.
+	IsDirectory bool
```

Comment:
I know it's simpler to have this as a member flag, but it does feel a bit icky, knowing that someone could pass a `file` and set `IsDirectory` to `true`, I'd rather it was checked which one it is right before generating the command.

I'm not expecting a change here, just curious to see others' opinion.

Reply:
Checking which one it is is only easily possible when diffing unstaged changes. In all other cases you'd have to get the information from git instead of from the file system (using something like `git ls-tree -r` in case it's a commit). Totally not worth it if you ask me, as the callers already know exactly what it is.

Besides:

> knowing that someone could pass a file and set IsDirectory to true

That's a totally valid and meaningful case, command-line git allows it, so if a caller really wants to do that, they should be allowed to.